apps/total-typescript/src/pages/invoices/[merchantChargeId].tsx

```diff
@@ -46,14 +58,26 @@ export const getServerSideProps: GetServerSideProps = async ({
         where: {id: purchase?.productId},
       })
 
+      if (product === null) {
+        return {
+          redirect: {
+            destination: '/invoices',
+            permanent: false,
+          },
+        }
+      }
+
       res.setHeader('Cache-Control', 's-maxage=1, stale-while-revalidate')
+
+      const props: InvoiceProps = {
+        charge,
+        product: convertToSerializeForNextResponse(product),
+        merchantChargeId,
+        bulkCoupon: convertToSerializeForNextResponse(bulkCoupon),
+      }
+
       return {
-        props: {
-          charge,
-          product: convertToSerializeForNextResponse(product),
-          merchantChargeId,
-          bulkCoupon: convertToSerializeForNextResponse(bulkCoupon),
-        },
+        props,
```

Comment:
If I declare `props` like this (with its explicit type as `InvoiceProps`) and then return it, I get type warnings right here for the specific values that have type issues. If I assemble and return props inline without explicitly referencing the type, then the type errors show up on the function and are an indecipherable bundle.